.github/workflows/ci.yml

```diff
@@ -77,61 +116,292 @@ jobs:
 
   windows:
     runs-on: windows-latest
+    permissions:
+      id-token: write # This is required for requesting the JWT
     steps:
       - name: Build ${{ env.PACKAGE_NAME }} + consumers
         run: |
           md D:\a\work
           cd D:\a\work
           python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz', 'builder.pyz')"
           python builder.pyz build -p ${{ env.PACKAGE_NAME }}
+      - name: configure AWS credentials (PubSub)
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ env.CI_PUBSUB_ROLE }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+      - name: run PubSub sample
+        run: |
+          cd D:\a\work
+          python -m pip install boto3
+          python ./aws-iot-device-sdk-cpp-v2/utils/run_sample_ci.py --language CPP --sample_file '.\aws-iot-device-sdk-cpp-v2\build\samples\pub_sub\basic_pub_sub\RelWithDebInfo\basic-pub-sub.exe' --sample_region ${{ env.AWS_DEFAULT_REGION }} --sample_secret_endpoint 'ci/endpoint' --sample_secret_certificate 'ci/PubSub/cert' --sample_secret_private_key 'ci/PubSub/key'
+      - name: configure AWS credentials (Device Advisor)
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ env.CI_DEVICE_ADVISOR }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+      - name: run DeviceAdvisor
+        run: |
+          cd D:\a\work\aws-iot-device-sdk-cpp-v2
+          python3 ./deviceadvisor/script/DATestRun.py
 
   windows-vs14:
     runs-on: windows-2019 # windows-2019 is last env with Visual Studio 2015 (v14.0)
     strategy:
       matrix:
         arch: [Win32, x64]
+      fail-fast: false
+    permissions:
+      id-token: write # This is required for requesting the JWT
     steps:
       - name: Build ${{ env.PACKAGE_NAME }} + consumers
         run: |
           md D:\a\work
```

Comment:
It would be nice if the D:\a\work could be turned into an environment variable that's set once at the top for easier mutability if needed.